observability/src/main/java/io/grpc/observability/logging/GcpLogSink.java

```diff
@@ -0,0 +1,147 @@
+import com.google.cloud.logging.Severity;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.protobuf.util.JsonFormat;
+import io.grpc.internal.JsonParser;
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Sink for Google Cloud Logging.
+ */
+public class GcpLogSink implements Sink {
+  private final Logger logger = Logger.getLogger(GcpLogSink.class.getName());
+
+  private static final String SERVICE_TO_EXCLUDE = "google.logging.v2.LoggingServiceV2";
+  private static final String DEFAULT_LOG_NAME = "grpc";
+  private static volatile GcpLogSink gcpLogSinkInstance = null;
+  private final Logging gcpLoggingClient;
+
+  /**
+   * Constructor for custom sink.
+   */
+  private GcpLogSink(String projectId) {
+    LoggingOptions.Builder builder = LoggingOptions.newBuilder();
+    if (!Strings.isNullOrEmpty(projectId)) {
+      builder.setProjectId(projectId);
+    }
+    gcpLoggingClient =  builder.build().getService();
+  }
+
+  private GcpLogSink(Logging client) {
+    gcpLoggingClient = client;
+  }
+
+  @VisibleForTesting
+  public static void setInstance(Logging loggingClient) {
+    gcpLogSinkInstance = new GcpLogSink(loggingClient);
+  }
+
+  /**
+   * Retrieves a single instance of GcpLogSink.
+   *
+   * @param destinationProjectId cloud project id to write logs
+   */
+  public static GcpLogSink getInstance(String destinationProjectId) {
+    if (gcpLogSinkInstance == null) {
+      synchronized (GcpLogSink.class) {
+        if (gcpLogSinkInstance == null) {
+          gcpLogSinkInstance = new GcpLogSink(destinationProjectId);
+        }
+      }
+    }
+    return gcpLogSinkInstance;
+  }
+
+  @Override
+  public synchronized void write(GrpcLogRecord logProto) {
+    if (gcpLoggingClient == null) {
+      logger.log(Level.SEVERE, "Attempt to write after GcpLogSink is closed.");
+      return;
+    }
+    if (SERVICE_TO_EXCLUDE.equals(logProto.getServiceName())) {
+      return;
+    }
+    try {
+      Map<String, Object> mapPayload = protoToMapConverter(logProto);
+      Severity logEntrySeverity = getCloudLoggingLevel(logProto.getLogLevel());
+      // TODO(vindhyan): make sure all (int, long) values are not displayed as double
+      LogEntry grpcLogEntry =
+          LogEntry.newBuilder(JsonPayload.of(mapPayload))
+              .setSeverity(logEntrySeverity)
+              .setLogName(DEFAULT_LOG_NAME)
+              .setResource(MonitoredResource.newBuilder("global").build())
+              .build();
+      gcpLoggingClient.write(Collections.singleton(grpcLogEntry));
+    } catch (Exception e) {
+      logger.log(Level.SEVERE, "Caught exception while writing to cloud logging", e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> protoToMapConverter(GrpcLogRecord logProto)
```

Comment:
Is there a unit test for this? If this is tested indirectly in the test for `write` then it's okay